src/DDTrace/Configuration.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace DDTrace;
+
+use DDTrace\Configuration\AbstractConfiguration;
+
+/**
+ * DDTrace global configuration object.
+ */
+class Configuration extends AbstractConfiguration
+{
+    /**
+     * Whether or not distributed tracing is enabled globally.
+     *
+     * @return bool
+     */
+    public function isDistributedTracingEnabled()
```

Comment:
I'm stumped here a bit. one hand I really like that we're using statically analyzable code to fetch configuration.

OTOH, adding a method for every configuration option - I feel like it might be a bit of an overkill in the long run.

WDYT if AbstractConfiguration implemented also `VariableRegistry` interface. 
So that boolValue could be reimplemented here.
And we could use this configuration like so:

`$globalConfiguration->boolValue('distributed.tracing');`

Hmm. This also means that we would have split the configuration defaults from the implementation. Maybe to a `DefaultVariableRegistry` ?

I might be optimizing a bit for the future here, so just thinking out loud. Otherwise I really like this approach!

Reply:
Ehehehe the solution that you proposed it much more php-style, so I would be totally for that.
To be honest, I really like to have statically analysable methods and I don't think that they will be dozens.
So I still gives my vote to methods 😄 but totally opened to change it if you guys prefer 😄

Reply:
I have a feeling we'll end up with both, and it should be cool too. I.e. the big things like `isDistrubutedTracingEnabled()` is really good case for methods approach.

The small things like:
`agent.connection.timeout`

Reply:
TBH I am not a big fan of allowing both 😄 but I also acknowledge that the string approach is more in line with PHP approach. So let me add both the access mechanism...but I hope that everybody will use the explicit method call 😄.